internal/controllers/machinehealthcheck/machinehealthcheck_targets.go

```diff
@@ -183,6 +183,21 @@ func (t *healthCheckTarget) needsRemediation(logger logr.Logger, timeoutForMachi
 			nextCheckTimes = append(nextCheckTimes, nextCheck)
 		}
 	}
+
+	// check certificate expiration
+	if t.MHC.Spec.CertificatesMinExpiryDuration != nil {
+		// Check only if the certificate expiry information is available.
+		if t.Machine.Status.CertificatesExpiryDate != nil {
+			certificatesExpireAt := t.Machine.Status.CertificatesExpiryDate.Time
+			if now.Add(t.MHC.Spec.CertificatesMinExpiryDuration.Duration).After(certificatesExpireAt) {
+				expiresIn := certificatesExpireAt.Sub(now)
+				expiredInRoundedToMinutes := expiresIn.Round(time.Minute)
```

Comment:
Have you considered using utils to format this in a human friendly format (e.g. 30d), for sure kubectl uses some of them, but I have to do some research

Reply:
That would definitely be nice. I dont know where I can find it of the top of my head. I will look for it and use it here.
However, I don't think we need to block this PR for this item.

Reply:
The .String() func of Duration should do this automatically.